proposal/color-4-hwb.md

```diff
@@ -0,0 +1,455 @@
+system that Sass already uses internally.
+
+[color-4]: https://www.w3.org/TR/css-color-4/
+
+## Summary
+
+> This section is non-normative.
+
+This proposal defines a Sassified version of the `hwb()` color function
+added to [CSS Color Level 4][color-4] – along with relevant inspection and
+adjustment options. This function will only be available inside the `sass:color`
+module to avoid conflicts with the CSS syntax, and will be converted to more
+common color-name, hex, or `rgba()` syntax for output -- following the same
+logic as our current color functions.
+
+- New `color.hwb()` function describes colors in the sRGB colorspace using
+  `$hue` (defined identically to the `hsl()` "hue" value), along with
+  `$whiteness`, `$blackness`, and optional `$alpha` transparency.
+- New `color.whiteness()` and `color.blackness()` functions return the respective
+  values of `w` or `b` for a given color.
+- Existing `color.adjust()`, `color.scale()`, and `color.change()` functions will
+  accept additional `$whiteness` and `$blackness` parameters before the final
+  `$alpha` parameter.
+
+### Design Decisions
+
+Both `rgb/a()` and `hsl/a()` are available in the global namespace because both
+of these formats are part of a stable CSS spec, and we want to make any standard
+CSS representation of a color parse as a Sass color. However, although `hwb()`
+is defined in Color Level 4, it's not yet implemented by any browser. Sass
+policy is to avoid supporting any new CSS syntax until it's shipped in a real
+browser, so `hwb()` **will not** be available in the global namespace initially.
+Instead, it will appear in the `sass:color` namespace which is guaranteed to be
+forwards-compatible with future CSS changes.
+
+Because the `color.hwb()` function isn't currently intended to directly
+implement CSS's native `hwb()` function, it **will not** accept *special number
+string* or *special variable string* values that can only be resolved in CSS.
+However, for consistency with Sass's `rgb()` and `hsl()` functions it will
+support both space-delimited and comma-delimited arguments.
+
+### Future Designs
+
+It's likely that as CSS Color Level 4 matures, `hwb()` will be stabilized and
+supported in browsers in one form or another. At this point, Sass will likely
+add support for a global `hwb()` function that's compatible with its CSS usage,
+including supporting special number and variable strings. The details of this
+are left to a future proposal.
+
+## Procedures
+
+### Scaling a Number 
+
+This algorithm takes a number `number`, a value `factor`, and a number `max`.
+It's written "scale `<number>` by `<factor>` with a `max` of `<max>`". It
+returns a number with a value between 0 and `max` and the same units as
+`number`.
+
+* If `factor` isn't a number with unit `%` between `-100%` and `100%`
+  (inclusive), throw an error.
+
+* If `factor > 0%`, return `number + (max - number) * factor / 100%`.
```

Comment:
What happens if `max < number`?

Reply:
In practice it never will be, since `number` always comes from a color channel which is guaranteed to be between 0 and `max`. I'll add a note to that effect.